addons/hr_holidays/models/hr_leave.py

```diff
@@ -404,6 +404,13 @@ def _compute_from_holiday_status_id(self):
         # TODO: remove in master
         self.holiday_allocation_id = False
 
+    @api.depends('date_from', 'date_to')
+    def _compute_request_date_from_to(self):
+        for leave in self:
+            leave_tz = timezone(leave.tz)
+            leave.request_date_from = pytz.utc.localize(leave.date_from).astimezone(leave_tz)
+            leave.request_date_to = pytz.utc.localize(leave.date_to).astimezone(leave_tz)
```

Comment:
We shouldn't create compute methods if the associated field isn't set as computed.
Instead, maybe do something like this:
```py
def _update_request_date(self, from_to='from'):
        for leave in self:
            leave_tz = timezone(leave.tz)
            leave['request_date_' + from_to] = pytz.utc.localize(leave['date_' + from_to]).astimezone(leave_tz)    
```
And call that method if the fields are modified. :slightly_smiling_face:
The loop could also directly be in the write itself

Reply:
I was hesitant to do so because it would need to be called after the write has finished, which I didn't know how it would interact with protected fields and recomputation. We just want to update the request_date_from/to but don't want to force a recomputation of date_from/to.

Do you know how it would interact in cases such as these? I feel like it's a bit tricky since the write() method takes in one dictionary and can write it to several records at the same time

Reply:
Also yes it is reproducible also in 17.0

Reply:
I don't think it would create an issue to use `leave.request_date_from = ...` as long as you don't explicitly call the `write` method itself :)
Good to note as well that your compute creates a compute loop which is also something that could cause discrepancies, so it's to be avoided.
If possible, could you add to your commit/PR message some steps to reproduce the issue?
Thanks!

Reply:
The field update method calls write() on the model so I needed to have it be a `protecting` function. I added the fields that needed protecting as a parameter in case of customizations that don't want recomputed fields.

I also added a test for checking that request_date_from/to is updated properly